### PR TITLE
Analyze Yakindu SCT Qt timer leak

### PR DESCRIPTION
**Pull request by stmork**
Note: This Qt timer service is originally generated by Yakindu SCT and not modified, yet.

### CODE REVIEW

**Comment by rhauns** [comment]
Work in progress, need more time ...

File: .github/workflows/build.yml
Comment (rhauns) [inline .github/workflows/build.yml:1]: Hoping for a time window some day to get a short introduction about the github workflow feature.

File: SCT-Cpp.pro [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 SUBDIRS  = \
 	SctCppTimer \
-	SctCppUnit
+	SctQtTimer \
+	SctUnitTimer
 
-SctCppTimer.file = SctCppTimer/SctCppTimer.pro
-SctCppUnit.file  = SctCppTimer/SctCppUnit.pro
+SctCppTimer.file  = SctCppTimer/SctCppTimer.pro
+SctQtTimer.file   = SctCppTimer/SctQtTimer.pro
```
Comment (rhauns) [inline SCT-Cpp.pro:8]: Builds, but gives us the known shadow and override warnings for the generated code.  
How about adding ``-Wshadow`` and ``-Wsuggest-override``  to the compiler flags?

Comment (rhauns) [reply SCT-Cpp.pro:8]: Hm... just found out, that those flags are already set. But why do i still get the warnings?

File: SctCppTimer/src-lib/sc_qt_timerservice.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    SCTimer *timer = nullptr;
+
+    // retrieve the timer map for the state machine
+    QMap<sc::eventid, SCTimer*> *eventTimerMap = machineTimerMapMap.value(statemachine);
+
+    // retrieve and stop a timer if it already exists for the event.
+    if (eventTimerMap != nullptr) {
+
+        if (eventTimerMap->contains(event)) {
+           SCTimer *timer = eventTimerMap->value(event);
```
Comment (rhauns) [inline SctCppTimer/src-lib/sc_qt_timerservice.cpp:43]: Here is the known SCT Qt-Timer-Service-Bug.  
And strange enough, even ``-Wshadow`` ist set in the _SctQtTimer.pro_ the warning is issued during build:  
```
/home/rhauns/src/sctcpptimer/SctCppTimer/src-lib/sc_qt_timerservice.cpp:43: warning: declaration of ‘timer’ shadows a previous local [-Wshadow]
   43 |            SCTimer *timer = eventTimerMap->value(event);
      |                     ^~~~~
```

**Review by rhauns — APPROVED** [review]

File: .github/workflows/build.yml
Comment (stmork) [reply .github/workflows/build.yml:1]: https://docs.github.com/de/actions

File: SCT-Cpp.pro (hunk 1 above)
Comment (stmork) [reply SCT-Cpp.pro:8]: Setting the compiler flags enables the warnings.

**Pull request merged**